x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_edit.tsx

```diff
@@ -73,10 +73,25 @@ const cloneAndMigrateRule = (initialRule: Rule) => {
             initialRule.notifyWhen === RuleNotifyWhen.THROTTLE ? initialRule.throttle! : null,
         }
       : { summary: false, notifyWhen: RuleNotifyWhen.THROTTLE, throttle: initialRule.throttle! };
-    clonedRule.actions = clonedRule.actions.map((action) => ({
-      ...action,
-      frequency,
-    }));
+    clonedRule.actions = clonedRule.actions.map((action) =>
+      action.type === RuleActionTypes.SYSTEM
+        ? action
+        : {
+            ...action,
+            frequency,
+          }
+    );
+  } else {
+    // Migrate untyped actions
+    clonedRule.actions = clonedRule.actions.map(
+      (action: SanitizedRuleAction | Omit<SanitizedRuleAction, 'type'>) =>
+        'type' in action
+          ? action
+          : ({
+              ...action,
+              type: action.frequency ? RuleActionTypes.DEFAULT : RuleActionTypes.SYSTEM,
+            } as SanitizedRuleAction)
+    );
```

Comment:
Curious to know why we need to migrate actions for an example log?

Reply:
This PR also fixes some issues that are necessary to deploy system actions more widely. We have to add action types to the payload sent via the form form in order to make rules with system actions editable — it throws errors if we send back an untyped payload — but this also breaks editing pre-existing rules if we don't migrate their actions.

Reply:
Re: Zoom conversation about how type defs aren't supposed to be used in the API. They're not. This type gets stripped from `update` API requests before sending it to the server.

When creating a new action, the `type` property is automatically added, because the actions form has already done the lookup work to determine if it's a system action. This enables us to use the `isSystemAction` type guard in the `rewriteBodyRequest` functions for `create` and `update`, which determines whether or not to attempt to add frequency, group, and alerts_filter params to the request.

Adding this migration step allows us to keep this logic the same for both creating and editing rules.

Reply:
Figured out a fix that doesn't require changing the Rule Edit form. Reverting.